### ISSUE TITLE
Batch actions with versioned data sources

### ISSUE THREAD

**Issue by michalisAsp** [aws-amplify/amplify-js#6021]
Hi, this is more of a question.

According to [docs](https://docs.amplify.aws/lib/datastore/how-it-works/q/platform/android#writing-data-from-the-appsync-console
) we can use batch actions with DataStore. But, I am having trouble setting it up using custom resolvers. I keep getting this error. 
```{
  "data": {
    "createBooks": null
  },
  "errors": [
    {
      "path": [
        "createBooks"
      ],
      "data": null,
      "errorType": "MappingTemplate",
      "errorInfo": null,
      "locations": [
        {
          "line": 4,
          "column": 2,
          "sourceName": null
        }
      ],
      "message": "Unsupported operation 'BatchPutItem'. Datasource Versioning only supports the following operations (TransactGetItems,PutItem,BatchGetItem,Scan,Query,GetItem,DeleteItem,UpdateItem,Sync)"
    }
  ]
}
```

Perhaps this is not supported? If not which batch actions are?

**Comment by michalisAsp** [comment]
> DataStore is designed primarily for developers to not have to focus on the backend and let your application code and workflow create everything. However, there will be some use cases where you will use the AppSync console, a Lambda function, or other out of band processes to write data (**such as batch actions** or data migrations) and you might send GraphQL operations without the DataStore client

@mauerbac Thanks but I see now that what I wrote might not be clear. So, let me try again.
According to the docs you can use batch actions to write data when using versioned enabled tables(DataStore is using those).
 But the error logs in my case(and probably in general) say that we can't. I think it would be useful if the the docs made this clear and gave options. E.g. write directly to DynamoDB...
```
"message": "Unsupported operation 'BatchPutItem'. Datasource Versioning only supports the following operations (TransactGetItems,PutItem,BatchGetItem,Scan,Query,GetItem,DeleteItem,UpdateItem,Sync)"

``` 

There are already these related issues I believe asking how to do something like this.
[3518](https://github.com/aws-amplify/amplify-cli/issues/3518)
[4449](https://github.com/aws-amplify/amplify-cli/issues/4449)

**Comment by mauerbac** [comment]
Hi @michalisAsp - ah, thanks for the clarification. I did misunderstand. Datastore core functionality is the same across all platforms but as you know the implementation does vary. I see you are using the Android implementation and the docs aren't super clear around the batch operation. I will move back into docs repo and tagging @jpignata who can help prioritize a docs update.  

Appreciate you tagging the other issues :)
